tachyon/math/polynomials/univariate/lagrange_interpolation.h

```diff
@@ -101,7 +101,7 @@ bool LagrangeInterpolate(
       coeffs_sums[0][j] += coeffs_sums[i][j];
     }
   }
-  *ret = Poly(Coeffs(std::move(coeffs_sums[0])));
+  *ret = Poly(Coeffs(std::move(coeffs_sums[0]), true));
```

Comment:
What about cases like
1. https://github.com/kroma-network/tachyon/blob/e182278d97c58c934c3d81f00aa710df9c1c5115/tachyon/math/polynomials/univariate/lagrange_interpolation.h#L34
and
2. https://github.com/kroma-network/tachyon/blob/e182278d97c58c934c3d81f00aa710df9c1c5115/tachyon/zk/plonk/vanishing/vanishing_prover_impl.h#L178
?
I'm still not really convinced with "enforcing clean construction of polynomials.

Reply:
As I understood, some operations rely on input polynomial to be _clean_. 
For example, https://github.com/kroma-network/tachyon/blob/e182278d97c58c934c3d81f00aa710df9c1c5115/tachyon/math/polynomials/univariate/univariate_sparse_coefficients.h#L227-L234

I think enforcing it will be more bug-free in the future. As for the cases you mentioned, I could recheck everything and pass `true` in the needed parts. I thought of making the dense polynomial to cleanup by default but still in your second case, we will have to pass `true` since the polynomial can be sparse.

Reply:
For now I added `true` to the cases you mentioned. I couldn't find any other instance.